runtime-sdk/src/dispatcher.rs

```diff
@@ -467,8 +447,148 @@ impl<R: Runtime + Send + Sync> transaction::dispatcher::Dispatcher for Dispatche
             messages,
             block_tags: block_tags.into_tags(),
             batch_weight_limits: Some(block_weight_limits),
+            tx_reject_hashes: vec![],
         })
     }
+}
+
+impl<R: Runtime + Send + Sync> transaction::dispatcher::Dispatcher for Dispatcher<R> {
+    fn execute_batch(
+        &self,
+        rt_ctx: transaction::Context<'_>,
+        batch: &TxnBatch,
+    ) -> Result<ExecuteBatchResult, RuntimeError> {
+        self.execute_batch_common(
+            rt_ctx,
+            |ctx| -> Result<Vec<ExecuteTxResult>, RuntimeError> {
+                // If prefetch limit is set enable prefetch.
+                let prefetch_enabled = R::PREFETCH_LIMIT > 0;
+
+                let mut txs = Vec::with_capacity(batch.len());
+                let mut prefixes: BTreeSet<Prefix> = BTreeSet::new();
+                for tx in batch.iter() {
+                    let tx_size = tx.len().try_into().map_err(|_| {
+                        Error::MalformedTransactionInBatch(anyhow!("transaction too large"))
+                    })?;
+                    // It is an error to include a malformed transaction in a batch. So instead of only
+                    // reporting a failed execution result, we fail the whole batch. This will make the compute
+                    // node vote for failure and the round will fail.
+                    //
+                    // Correct proposers should only include transactions which have passed check_tx.
+                    let tx = Self::decode_tx(ctx, tx)
+                        .map_err(|err| Error::MalformedTransactionInBatch(err.into()))?;
+                    txs.push((tx_size, tx.clone()));
+
+                    if prefetch_enabled {
+                        Self::prefetch_tx(&mut prefixes, tx)?;
+                    }
+                }
+                if prefetch_enabled {
+                    ctx.runtime_state()
+                        .prefetch_prefixes(prefixes.into_iter().collect(), R::PREFETCH_LIMIT);
+                }
+
+                // Execute the batch.
+                let mut results = Vec::with_capacity(batch.len());
+                for (index, (tx_size, tx)) in txs.into_iter().enumerate() {
+                    results.push(Self::execute_tx(ctx, tx_size, tx, index)?);
+                }
+
+                Ok(results)
+            },
+        )
+    }
+
+    fn schedule_and_execute_batch(
```

Comment:
Is this supposed to be called from somewhere in this PR?

Reply:
No, this is part of the dispatcher trait from Oasis Core, see the PR in https://github.com/oasisprotocol/oasis-core/pull/4446.